robottelo/hosts.py

```diff
@@ -479,6 +480,119 @@ def install_cockpit(self):
+        :param setup_remote_execution_pull: Deploy pull provider client on host
+        :param lifecycle_environment: Lifecycle environment.
+        :param operating_system: Operating system.
+        :param packages: A list of packages to install on the host when registered.
+        :param repo: Repository to be added before the registration is performed, supply url.
+        :param repo_gpg_key_url: Public key to verify the package signatures, supply url.
+        :param remote_execution_interface: Identifier of the host interface for remote execution.
+        :param update_packages: Update all packages on the host.
+        :param ignore_subman_errors: Ignore subscription manager errors.
+        :param force: Register the content host even if it's already registered.
+        :param insecure: Don't verify server authenticity.
+        :param username: Satellite admin username
+        :param password: Satellite admin password
+        :return: SSHCommandResult instance filled with the result of the
+            registration.
+        """
+        insights = (
+            # requires OS repo enabled for host
+            f'&setup_insights={str(setup_insights).lower()}'
+            if setup_insights is not None
+            else ''
+        )
+        rex = (
+            f'&setup_remote_execution={str(setup_remote_execution).lower()}'
+            if setup_remote_execution is not None
+            else ''
+        )
+        rex_pull = (
+            # requires Satellite Client repo enabled for host
+            f'&setup_remote_execution_pull={str(setup_remote_execution_pull).lower()}'
+            if setup_remote_execution_pull is not None
+            else ''
+        )
+        lce = ''
+        if lifecycle_environment:
+            selected_lce = entities.LifecycleEnvironment().search(
+                query={'search': f'name="{lifecycle_environment}"'}
+            )[0]
+            lce = f'&lifecycle_environment_id={selected_lce.id}'
+        os = ''
+        if operating_system:
+            selected_os = entities.OperatingSystem().search(
+                query={'search': f'name="{operating_system}"'}
+            )[0]
+            os = f'&operating_system_id={selected_os.id}'
+        pkgs = f'&packages={"+".join(packages)}' if packages is not None else ''
+        rex_iface = (
+            f'&remote_execution_interface={remote_execution_interface}'
+            if remote_execution_interface is not None
+            else ''
+        )
+        rp = f'&{urlencode({"repo": repo })}' if repo is not None else ''
+        gpg = (
+            f'&{urlencode({"repo_gpg_key_url": repo_gpg_key_url })}'
+            if repo_gpg_key_url is not None
+            else ''
+        )
+        cmd = (
+            'curl -sS '
+            f'-u {username}:{password} '
+            f'{"--insecure " if insecure else ""}'
+            f"'https://{target.hostname}:9090/register?"
```

Comment:
Change here over the previous iteration, we need capsule port here, not the ssh port that is in target.port. Capsule port change is afaik not supported, therefore not worrying about the hardcode here